lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2297,6 +2299,182 @@ class ArchiveTool : public Tool {
   }
 };
 
+#pragma mark - StaleFileRemovalTool implementation
+
+class StaleFileRemovalCommand : public Command {
```

Comment:
One last missing piece: you need to override and implement `getSignature()` to include the extra fields here... see `ShellCommand` for an example.